src/otx/v2/adapters/torch/mmengine/dataset.py

```diff
@@ -0,0 +1,310 @@
+        """
+        super().__init__(
+            task,
+            train_type,
+            train_data_roots,
+            train_ann_files,
+            val_data_roots,
+            val_ann_files,
+            test_data_roots,
+            test_ann_files,
+            unlabeled_data_roots,
+            unlabeled_file_list,
+            data_format,
+        )
+        self.scope = "mmengine"
+        self.dataset_registry: Registry = MMEngineRegistry().get("dataset")
+
+    def _initialize(self) -> None:
+        self.set_datumaro_adapters()  # Set self.dataset_entity & self.label_schema
+        self.base_dataset = self._get_sub_task_dataset()
+        self.initialize = True
+
+    def _get_sub_task_dataset(self) -> TorchDataset:
+        """Returns a TorchDataset object containing the data for the current sub-task.
+
+        Returns:
+            TorchDataset: A TorchDataset object containing the data for the current sub-task.
+        """
+        raise NotImplementedError
+
+    def _build_dataset(
+        self,
+        subset: str,
+        pipeline: list | dict | None = None,
+        config: str | (dict | Config) | None = None,
+    ) -> TorchDataset | None:
+        """Builds a TorchDataset object for the given subset using the specified pipeline and configuration.
+
+        Args:
+            subset (str): The subset to build the dataset for.
+            pipeline (Optional[Union[list, dict]]): The pipeline to use for the dataset.
+                Defaults to None.
+            config (Optional[Union[str, dict, Config]]): The configuration to use for the dataset.
+                Defaults to None.
+
+        Returns:
+            Optional[TorchDataset]: The built TorchDataset object, or None if the dataset is empty.
+        """
+        if not self.initialize:
+            self._initialize()
+
+        if subset not in SUBSET_LIST:
+            msg = f"{subset} is not supported subset"
+            raise ValueError(msg)
+
+        otx_dataset = self.dataset_entity.get_subset(str_to_subset_type(subset))
+        labels = self.label_schema.get_labels(include_empty=False)
+        if len(otx_dataset) < 1:
+            return None
+        # Case without config
+        if config is None:
+            dataset = self.base_dataset(otx_dataset=otx_dataset, labels=labels, pipeline=pipeline)
+            dataset.configs = {
+                "type": str(self.base_dataset.__qualname__),
+                "data_root": getattr(self, f"{subset}_data_roots"),
+                "ann_file": getattr(self, f"{subset}_ann_files"),
+                "data_prefix": "",
+                "pipeline": pipeline,
+            }
+            return dataset
+
+        # Config Setting
+        if isinstance(config, str):
+            _config = Config.fromfile(filename=config)
+        elif isinstance(config, dict):
+            _config = Config(cfg_dict=config)
+        else:
+            _config = config
+        # Case with Config
```

Comment:
Can you elaborate these comments for future reference?

Reply:
This is an unnecessary comment. Cleared.